### ISSUE TITLE
lua function not execute on readonly slave node

### ISSUE THREAD

**Issue by nichliu** [luin/ioredis#197]
I noticed lua function not execute on readonly slave node
I used redis.defineCommand to define the lua function

I defined readonly nodes during the connection. 

Redis 3.05 cluster

**Comment by luin** [comment]
Could you please post your code here that can reproduce the problem?

**Comment by nichliu** [comment]
```javascript
var Redis = require('ioredis');

var r1 = "x.x.x.x";
var r1s = "x.x.x.x";
var r2 = "x.x.x.x";
var r2s = "x.x.x.x";
var r3 = "x.x.x.x";
var r3s = "x.x.x.x";
var r4 = "x.x.x.x";
var r4s = "x.x.x.x";

var redisNodes= [{
    port: 6379,
    host: r1
},{
    port: 6379,
    host: r2
},{
    port: 6379,
    host: r3
},{
    port:6379,
    host:r4
},{
    port:6379,
    host:r4s,
    readOnly: true
},{
    port:6379,
    host:r1s,
    readOnly: true
},{
    port:6379,
    host:r2s,
    readOnly: true
},{
    port:6379,
    host:r3s,
    readOnly: true
}];

var redis = new Redis.Cluster(redisNodes);

redis.defineCommand('calc', {
    numberOfKeys: 1,
    lua: '.....lua code'
});

redis.calc(key,start,end,screen,function(err,data){
    #process result or err
})

**Comment by luin** [comment]
The `readOnly` option should be defined in the second parameter:

```javascript
var cluster = new Redis.Cluster(nodes, { readOnly: true });
```

This option means every command sent from the `cluster` instance is readonly so that ioredis will send them to both masters and slaves instead of just masters in order to scale reads and writes.

I understand you intention that sending readonly commands to slaves and others to masters. This feature has been discussed in #170 and will be implemented in ioredis v2.0.

**Comment by nichliu** [comment]
Thx. I got it.
